notebooks/external_augmentation.ipynb

```diff
@@ -0,0 +1,247 @@
+      "\u001b[0;31mModuleNotFoundError\u001b[0m: No module named 'xy'"
+     ],
+     "ename": "ModuleNotFoundError",
+     "evalue": "No module named 'xy'",
+     "output_type": "error"
+    }
+   ],
+   "source": [
+    "# lets prepare a basic dataset (e.g. one from `torchvision`)\n",
+    "import os\n",
+    "import torchvision\n",
+    "import numpy as np\n",
+    "\n",
+    "def to_array(inp):\n",
+    "    \"\"\"\n",
+    "    We need a small helper in this example because torchvision datasets output PIL\n",
+    "    images. When using them in combination with `rising`,\n",
+    "    just add `torchvision.transforms.ToTensor()`to the transform of the dataset\n",
+    "    \n",
+    "    Returns\n",
+    "    -------\n",
+    "    numpy.ndarray\n",
+    "        converted data\n",
+    "    \"\"\"\n",
+    "    from PIL import Image\n",
+    "    if isinstance(inp, Image.Image):\n",
+    "        return np.array(inp, np.float32, copy=False)[None]\n",
+    "    else:\n",
+    "        return inp\n",
+    "\n",
+    "dataset = torchvision.datasets.MNIST(\n",
+    "    os.getcwd(), train=True, download=True, transform=to_array)"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n",
+     "is_executing": false
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "#  plot shape\n",
+    "print(dataset[0][0].shape)\n",
+    "# visualize a single image\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "plt.imshow(dataset[0][0][0], cmap='gray')\n",
+    "plt.colorbar()\n",
+    "plt.show()"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n",
+     "is_executing": false
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "# helper function to visualize batches of images\n",
+    "import torch\n",
+    "\n",
+    "def show_batch(batch: torch.Tensor):\n",
+    "    grid = torchvision.utils.make_grid(batch)\n",
+    "    plt.imshow(grid[0], cmap='gray')\n",
+    "    # plt.colorbar()\n",
+    "    plt.show()"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n",
+     "is_executing": false
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Integration of `Batchgenerator` transformations into the augmentation pipeline.\n",
+    "Note: when batchgenerator transformations are integrated, gradients can not be propagated through\n",
```

Comment:
The framework is called `batchgenerators` (we should make sure the name is spelled correct and link to it)